DAQ_FW/lib/IADCSensor/IADCSensor.cpp

```diff
@@ -0,0 +1,77 @@
+#include "IADCSensor.h"
+
+// ADC INFO
+// adc voltage 2V~5.5V, 860 sample rate, SERIAL output, 16 bit resolution, 15 µV min voltage increment
+// i2c interface
+// PGA offers input ranges from ±0.256 V to ±6.144 V
+// adc resolution : 65536
+
+// PGA VALUES for a ADS1115 -
+// TODO: MOVE TO SYS_CONFIG WHEN FINAL
+#define ADS1115_GAIN_TWOTHIRDS 0.187500f
+#define ADS1115_GAIN_ONE 0.125000f
+#define ADS1115_GAIN_TWO 0.062500f
+#define ADS1115_GAIN_FOUR 0.031250f
+#define ADS1115_GAIN_EIGHT 0.015625f
+#define ADS1115_GAIN_SIXTEEN 0.007813f
+
+// CLASS METHODS:
+
+// Default constructor with generic metadata
+IADCSensor::IADCSensor(const char *_SensorName, uint16_t _SensorID, uint16_t _ADC_ID, uint16_t _ADC_GAIN_MODE)
+    : mSensorName(_SensorName), mSensorID(_SensorID), mADC_ID(_ADC_ID)
+{
+    ADCcovert(_ADC_GAIN_MODE);
+
+    // obj declared when logger not active, could be hard to log
+    Logger::Notice("Class created with gain mode: %u and V/per bit of %D", _ADC_GAIN_MODE, mVOLTAGE_PER_BIT);
+}
+
+float IADCSensor::Process()
```

Comment:
`Const correctness` is just a C++ habit of labeling methods as const if they don't modify the member variables. This makes it easier to debug when you find that member variables are changing unexpectedly; instead of checking every method, you can ignore ones labeled as const.

As good practice, you should try and figure out if your method will be const when you first declare them as it gives you a stricter guideline when actually writing the implementation. If you accidentally modify a member variable you'll get a compile-time error.

More info here: https://stackoverflow.com/questions/136880/sell-me-const-correctness 

TLDR: This method I think should be const